Source/Plugins/Core/com.equella.core/js/tsrc/search/components/SearchResultAttachmentsList.tsx

```diff
@@ -0,0 +1,342 @@
+    uuid,
+    version,
+    displayOptions,
+    keywordFoundInAttachment,
+    attachments = [],
+  },
+  getViewerDetails,
+  handleError,
+}: SearchResultAttachmentsListProps) => {
+  const itemKey = `${uuid}/${version}`;
+
+  const classes = useStyles();
+  const inSelectionSession: boolean = isSelectionSessionOpen();
+  const inSkinny = isSelectionSessionInSkinny();
+  const inStructured = isSelectionSessionInStructured();
+
+  const [attachExpanded, setAttachExpanded] = useState(
+    (inSelectionSession
+      ? displayOptions?.integrationOpen
+      : displayOptions?.standardOpen) ?? false
+  );
+
+  const [attachmentsAndViewers, setAttachmentsAndViewers] = useState<
+    AttachmentAndViewer[]
+  >([]);
+
+  // Responsible for determining the MIME type viewer for the provided attachments
+  useEffect(() => {
+    let mounted = true;
+
+    if (!attachments.length) {
+      // If there are no attachments, skip this effect
+      return;
+    }
+
+    const transform = async (
+      attachment: OEQ.Search.Attachment
+    ): Promise<AttachmentAndViewer> => {
+      const {
+        attachmentType,
+        mimeType,
+        links: { view: defaultViewUrl },
+        filePath,
+      } = attachment;
+      const viewUrl = determineAttachmentViewUrl(
+        uuid,
+        version,
+        attachmentType,
+        defaultViewUrl,
+        filePath
+      );
+
+      let viewerDetails: OEQ.MimeType.MimeTypeViewerDetail | undefined;
+      try {
+        viewerDetails = mimeType ? await getViewerDetails(mimeType) : undefined;
+      } catch (error) {
+        handleError({
+          ...error,
+          message: `${searchResultStrings.errors.getAttachmentViewerDetailsFailure}: ${error.message}`,
+        });
+      }
+
+      return {
+        attachment: attachment,
+        viewer: determineViewer(
+          attachmentType,
+          viewUrl,
+          mimeType,
+          viewerDetails?.viewerId
+        ),
+      };
+    };
+
+    (async () => {
+      const attachmentsAndViewers = await Promise.all(
+        attachments.map<Promise<AttachmentAndViewer>>(transform)
+      );
+      if (mounted) {
+        setAttachmentsAndViewers(attachmentsAndViewers);
+      }
+    })();
+
+    return () => {
+      // Short circuit if this component is unmounted before all its comms are done.
+      mounted = false;
+    };
+  }, [attachments, getViewerDetails, handleError, uuid, version]);
+
+  const handleSelectResource = (
+    itemKey: string,
+    attachments: string[] = []
+  ) => {
+    selectResource(itemKey, attachments).catch(handleError);
+  };
+
+  const handleAttachmentPanelClick = (event: SyntheticEvent) => {
+    /** prevents the SearchResult onClick from firing when attachment panel is clicked */
+    event.stopPropagation();
+    setAttachExpanded(!attachExpanded);
+  };
+
+  const lightboxAttachments = attachmentsAndViewers.filter(
+    (av) => av.viewer[0] === "lightbox"
+  );
+  const buildLightboxNavigationHandler = (
```

Comment:
Please have a blank line around definitions of functions.

```suggestion

  const buildLightboxNavigationHandler = (
```